### ISSUE TITLE
Replace all Arrays with Vectors

### ISSUE THREAD

**Issue by moly** [FlixelCommunity/flixel#104]
Theoretically flixel should run quicker if everything used typed Vectors instead of Arrays. This would probably need to be benchmarked to confirm.

**Comment by IQAndreas** [comment]
Last I heard Arrays are only faster for <can't remember name> types, such as `int`, `uint`, `Number`, `Boolean`, and `String`. Otherwise Arrays are pretty much identical in speed to Vectors.

EDIT: I'm not sure if these stats hold for the current version of Flash Player, but Vectors for all other data types have faster read speed, but slower write speed. But considering how 90% of the time you are reading from them, I think this difference is worth it.
http://jacksondunstan.com/articles/649

But I'm always a fan of stricter typing, so I am all for using Vectors instead.

**Comment by Dovyski** [comment]
I was going to post that same Jackson's article :)

I think we could run his benchmark using the current Flash Player and check if the numbers are still valid. After we can make some tests to check if Flixel would benefit from using `Vector` instead of `Array`.

**Comment by IQAndreas** [comment]
@Dovyski Did you run those tests? Even if Vectors are (for example) 10% slower than Arrays, I still think strict typing is a better idea, and prevents more problems in the future.

If the performance difference isn't huge, shall we go ahead and include this (or at least part of it) in the current version (Flixel 2.58)?

**Comment by Dovyski** [comment]
No, I didn't. I'll check on that as soon as I have some free time. I'm ok about adding such change to the current version. Strict typing FTW!
